### PR TITLE
Add /proc/net/wireless parsing

### LINKED ISSUE #479: Add support for `/proc/net/wireless` parsing
To collect wireless RSSI values using procfs. We can also add this support to node_exporter once we implement this.

Example:
```
$ cat /proc/net/wireless

Inter-| sta-|   Quality        |   Discarded packets               | Missed | WE
 face | tus | link level noise |  nwid  crypt   frag  retry   misc | beacon | 22

wlo1: 0000   70.  -31.  -256        0      0      0      0     25        0
```

/assign

### PR DESCRIPTION
**Pull request by Dentrax**
Fixes #479

cc @discordianfish

Signed-off-by: Furkan <furkan.turkal@trendyol.com>

### CODE REVIEW

**Comment by Dentrax** [comment]
@discordianfish PTAL when possible? Would really appreciate if you can check this and give a go ahead if all is well.

File: net_wireless.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+		}
+
+		name := strings.TrimSpace(parts[0])
+		stats := strings.Fields(parts[1])
+
+		if len(stats) < 10 {
+			return nil, fmt.Errorf("invalid number of fields in line %d, expected at least 10, got %d: %q", n, len(stats), line)
+		}
+
+		status, err := strconv.Atoi(stats[0])
```
Comment (SuperQ) [inline net_wireless.go:110]: The status field appears to be printed as a a 4-digit hex value. Not an integer.

See: https://github.com/torvalds/linux/blob/a5c95ca18a98d742d0a4a04063c32556b5b66378/net/wireless/wext-proc.c#L49-L50

```C
		seq_printf(seq, "%6s: %04x  %3d%c  %3d%c  %3d%c  %6d %6d %6d "
				"%6d %6d   %6d\n",
			   dev->name, stats->status, stats->qual.qual,
			   stats->qual.updated & IW_QUAL_QUAL_UPDATED
			   ? '.' : ' ',
			   ((__s32) stats->qual.level) -
			   ((stats->qual.updated & IW_QUAL_DBM) ? 0x100 : 0),
			   stats->qual.updated & IW_QUAL_LEVEL_UPDATED
			   ? '.' : ' ',
			   ((__s32) stats->qual.noise) -
			   ((stats->qual.updated & IW_QUAL_DBM) ? 0x100 : 0),
			   stats->qual.updated & IW_QUAL_NOISE_UPDATED
			   ? '.' : ' ',
			   stats->discard.nwid, stats->discard.code,
			   stats->discard.fragment, stats->discard.retries,
			   stats->discard.misc, stats->miss.beacon);
```

The value comes from https://github.com/torvalds/linux/blob/master/include/uapi/linux/wireless.h#L881-L8y89.

```C
/* ------------------------ WIRELESS STATS ------------------------ */
/*
 * Wireless statistics (used for /proc/net/wireless)
 */
struct iw_statistics {
	__u16		status;		/* Status
					 * - device dependent for now */

	struct iw_quality	qual;		/* Quality of the link
						 * (instant/mean/max) */
	struct iw_discarded	discard;	/* Packet discarded counts */
	struct iw_missed	miss;		/* Packet missed counts */
};
```

Comment (Dentrax) [reply net_wireless.go:110]: > The status field appears to be printed as a a 4-digit hex value. Not an integer.

Oh, nice catch. In this case, should I return a `float` type? Or maybe `[]byte`?

Comment (SuperQ) [reply net_wireless.go:110]: I think `strconv.ParseUint(stats[0], 16, 16)` will be fine.

Comment (Dentrax) [reply net_wireless.go]: Thanks! Ack. Updated the test.

**Review by SuperQ — APPROVED** [review]
LGTM

**Review by pgier — APPROVED** [review]
LGTM

**Pull request merged**